### PR TITLE
global: fix fields sorting in details, create and edit views

### PR DESCRIPTION
**Pull request by anikachurilova**
* closes https://github.com/inveniosoftware/invenio-administration/issues/155
* requires https://github.com/inveniosoftware/invenio-rdm-records/pull/1193

Before: fields order in details, create and edit views is alphabetical
After: fields order corresponds to the one specified in [config](https://github.com/inveniosoftware/invenio-banners/blob/master/invenio_banners/administration/banners.py).

Details page:
Screenshot 2023-02-16 at 14 26 51

Create page:
Screenshot 2023-02-16 at 14 42 06

Edit page:
Screenshot 2023-02-16 at 14 42 42

### CODE REVIEW

File: invenio_administration/assets/semantic-ui/js/invenio_administration/src/formik/GenerateForm.js [hunk 1]
```diff
@@ -69,6 +70,15 @@ const mapFormFields = (obj, parentField, isCreate, formFieldsConfig, dropDumpOnl
     return null;
   }
 
+  // copy order field from config to marshmallow schema
```
Comment (jrcastro2) [inline invenio_administration/assets/semantic-ui/js/invenio_administration/src/formik/GenerateForm.js:73]: Question: Why do we need this?

Comment (anikachurilova) [reply invenio_administration/assets/semantic-ui/js/invenio_administration/src/formik/GenerateForm.js:73]: The `obj` field - is a marshmallow schema, which doesn't know anything about the order. We still have to use it in further actions, so the idea was to just copy `order` field from config there.

File: invenio_administration/assets/semantic-ui/js/invenio_administration/src/components/utils.js [hunk 2]
```diff
@@ -9,8 +9,8 @@ export const sortFields = (schema) => {
     Object.entries(schema)
       // sort by order
       .sort((a, b) => {
-        if (a[1]?.ui && b[1].ui) {
-          return a[1].ui.order > b[1].ui.order;
+        if (a[1] && b[1]) {
+          return a[1].order - b[1].order;
```
Comment (alejandromumo) [inline invenio_administration/assets/semantic-ui/js/invenio_administration/src/components/utils.js:13]: Is this change needed?

File: invenio_administration/assets/semantic-ui/js/invenio_administration/src/formik/GenerateForm.js (hunk 1 above)
Comment (alejandromumo) [reply invenio_administration/assets/semantic-ui/js/invenio_administration/src/formik/GenerateForm.js:73]: Fields `order` should come from the backend (as it was before with `resourceSchema`). Conceptually, at some point the backend should deserialize `uiSchema` taking into account `ui` parameters found in `resourceSchema` (if any).

However, I do not understand what was failing here. `resourceSchema` was not passing an `ui` property anymore? That's the only reason I can see why this was failing. The sort function is conceptually the same (sorting by `>` or `-` yields the same boolean value needed by `sort`)

File: invenio_administration/assets/semantic-ui/js/invenio_administration/src/components/utils.js (hunk 2 above)
Comment (anikachurilova) [reply invenio_administration/assets/semantic-ui/js/invenio_administration/src/components/utils.js:17]: Yes, it's a sorting using a comparison function. See:https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#:~:text=copy%20is%20made.-,Description,-If%20compareFn%20is

File: invenio_administration/assets/semantic-ui/js/invenio_administration/src/formik/GenerateForm.js (hunk 1 above)
Comment (alejandromumo) [reply invenio_administration/assets/semantic-ui/js/invenio_administration/src/formik/GenerateForm.js:73]: Discussed IRL about this.

It seems that, [at some point](https://github.com/inveniosoftware/invenio-administration/commit/794ed8532fd02ba80126e1f479c4acfc310782c6#diff-b7a1171262f96ab548ab96789d64e3003b17f599f9a3ab8fb97a0f4781513afdL239), we separated `resourceSchema` from `uiSchema`. However, the `sort` function expects that the schema has a property `.ui` that was previously removed (see previous commit).
Moreover, the `sortFields` function reducer is wrong. It's reducing an array to an object and objects in JS are indexed by key (using its hash), therefore everything seems to be "ordered" alphabetically. Changing this means there are a couple of small changes down the  way.

Comment (anikachurilova) [reply invenio_administration/assets/semantic-ui/js/invenio_administration/src/formik/GenerateForm.js]: Thank you for your comments!
In the end, removed copying fields from one schema to another and changed with using necessary schema in a corresponding place: ui schema for rendering, marshmallow schema - for de/serialisation.
Thanks to @alejandromumo for support and help!

File: invenio_administration/assets/semantic-ui/js/invenio_administration/src/components/utils.js (hunk 2 above)
Comment (anikachurilova) [reply invenio_administration/assets/semantic-ui/js/invenio_administration/src/components/utils.js:13]: Thanks for pointing this out!
It looks like the `a - b` and the `a > b` comparison functions do the same thing, however, the `sort()` function excepts a comparison function, which has to return a number, otherwise it performs a cast to a number, which doesn't look like we expect. See the link: https://stackoverflow.com/questions/24080785/sorting-in-javascript-shouldnt-returning-a-boolean-be-enough-for-a-comparison (both answers are useful).

**Review by kpsherva — APPROVED** [review]
